pkg/supportbundle/spec.go

```diff
@@ -0,0 +1,740 @@
+			builtBundle.Spec.Analyzers = kotsKinds.Analyzer.DeepCopy().Spec.Analyzers
+		}
+	}
+
+	app, err := store.GetStore().GetApp(appID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get app")
+	}
+
+	err = injectDefaults(app, builtBundle, opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to inject defaults")
+	}
+
+	s := serializer.NewYAMLSerializer(serializer.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+	var b bytes.Buffer
+	if err := s.Encode(builtBundle, &b); err != nil {
+		return nil, errors.Wrap(err, "failed to encode support bundle")
+	}
+
+	templatedSpec := b.Bytes()
+
+	renderedSpec, err := helper.RenderAppFile(app, &sequence, templatedSpec, kotsKinds)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed render support bundle spec")
+	}
+
+	// unmarshal the spec, look for image replacements in collectors and then remarshal
+	// we do this after template rendering to support templating and then replacement
+	supportBundle, err := kotsutil.LoadSupportBundleFromContents(renderedSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal rendered support bundle spec")
+	}
+
+	registrySettings, err := store.GetStore().GetRegistryDetailsForApp(appID)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get registry settings for app")
+	}
+
+	collectors, err := registry.UpdateCollectorSpecsWithRegistryData(supportBundle.Spec.Collectors, registrySettings, kotsKinds.Installation.Spec.KnownImages, kotsKinds.License)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to update collectors")
+	}
+	supportBundle.Spec.Collectors = collectors
+	b.Reset()
+	if err := s.Encode(supportBundle, &b); err != nil {
+		return nil, errors.Wrap(err, "failed to encode support bundle")
+	}
+
+	renderedSpec = b.Bytes()
+
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get k8s clientset")
+	}
+
+	secretName := GetSpecSecretName(app.Slug)
+
+	existingSecret, err := clientset.CoreV1().Secrets(os.Getenv("POD_NAMESPACE")).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil && !kuberneteserrors.IsNotFound(err) {
+		return nil, errors.Wrap(err, "failed to read support bundle secret")
+	} else if kuberneteserrors.IsNotFound(err) {
+		secret := &corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "v1",
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: os.Getenv("POD_NAMESPACE"),
+				Labels:    kotstypes.GetKotsadmLabels(),
+			},
+			Data: map[string][]byte{
+				SpecDataKey: renderedSpec,
+			},
+		}
+
+		_, err = clientset.CoreV1().Secrets(os.Getenv("POD_NAMESPACE")).Create(context.TODO(), secret, metav1.CreateOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create support bundle secret")
+		}
+
+		return supportBundle, nil
+	}
+
+	if existingSecret.Data == nil {
+		existingSecret.Data = map[string][]byte{}
```

Comment:
Map should be initialized here so we don't panic when adding a key on 152.

`existingSecret.Data = make(map[string][]byte{})`

Reply:
Looked into this more and actually the braces at the end are initializing and empty map literal. Modified this example from Tour of Go https://goplay.space/#QuoyfUJ44pY. I don't think we have a style guide and more than likely I copied from another location, so I'm inclined to leave as-is.